### PR TITLE
[SMB] Add --enum-shares options like in NFS protocol

### PR DESCRIPTION
**Pull request by y0no**
This PR give the possibility to enum SMB shares like with NFS.

```
poetry run NetExec smb 192.168.56.11 -u Administrator -p  "<REDACTED>" --enum-shares 1 
```

image

### CODE REVIEW

**Comment by mpgn** [comment]
Thx for the pr, but i think it's better use spider_plus modules imo

**Comment by y0no** [comment]
@mpgn oh you're right ! I hadn't thought  about spider_plus !

In fact I'm able to reproduce the same behavior with "--spider" options : 

```
poetry run NetExec smb 192.168.56.11 -u Administrator -p  "<REDACTED" --spider C$ --depth 1 --pattern ''
```
image

The view just lack of information about the permissions on each files (maybe I could add it?) 

But the "spider_plus" module, give me this error on "GOAD Light" lab: 

```
poetry run NetExec smb 192.168.56.11 -u Administrator -p  "<REDACTED>" -M spider_plus  -o EXCLUDE_FILTER=print$,ipc$,admin$
```

image

Am I missing something ? In addition, I don't see options on this module to set a depth for the spidering, is it compatible with "--depth" ?

**Comment by y0no** [comment]
Hi @NeffIsBack, thank you for your feedback, so you ask for a syntax like `--dir "C$/Documents and Settings"` ?
With the code below I can quickly implement this feature and keep the output like in the first screenshot.

**Comment by NeffIsBack** [comment]
Yep exactly, but for the share we should use the already implemented `--share` flag

**Comment by y0no** [comment]
I have implemented the `--dir` option, with this syntax:

```
poetry run NetExec smb 192.168.56.11 -u Administrator -p  "<REDACTED>" --dir "Users\Vagrant"
```
image

The share can be defined as mentionned with `--share` argument. Is it ok for you ?

**Comment by NeffIsBack** [comment]
Yep, looks good! Nice one, thanks for the PR :)

File: nxc/protocols/smb.py [hunk 1]
```diff
@@ -903,6 +903,34 @@ def shares(self):
             self.logger.highlight(f"{name:<15} {','.join(perms):<15} {remark}")
         return permissions
 
+
+    def dir(self):
+        # Seems defined by default, do we have to keep this check ?
+        if not self.args.share:
+            self.logger.error("You must define --share option")
+            return
```
Comment (NeffIsBack) [inline nxc/protocols/smb.py:911]: You were right with that, that check isn't necessarily needed

**Review by NeffIsBack — COMMENTED** [review]
If you have the time you could remove the check. Will test it soon and merge accordingly :)

File: nxc/protocols/smb.py (hunk 1 above)
Comment (NeffIsBack) [reply nxc/protocols/smb.py]: Fixed that quick

**Comment by NeffIsBack** [comment]
@y0no forgot to mention, it would be really great if you could also PR documentation for the new flag to our wiki repo:
https://github.com/Pennyw0rth/NetExec-Wiki

**Pull request merged**